### PR TITLE
Dev Container Lockfile

### PR DESCRIPTION
**Pull request by chrmarti**

### CODE REVIEW

File: proposals/devcontainer-lockfile.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+```jsonc
+{
+    "features": {
+        "ghcr.io/devcontainers/features/node:1": {
+            "version": "1.0.4",
+            "resolved": "ghcr.io/devcontainers/features/node@sha256:567d704b3f4d3eca3acee51ded7c460a8395436d135d53d1175fb565daff42b8",
+            "integrity": "sha256:567d704b3f4d3eca3acee51ded7c460a8395436d135d53d1175fb565daff42b8"
+        },
+        "https://mycustomdomain.com/myscript-1.2.3.tgz": {
```
Comment (joshspicer) [inline proposals/devcontainer-lockfile.md:35]: The packaging tooling would not generate a tgz with this name. In fact, the tooling expects a certain format to parse out the name prior to download the tarball

https://containers.dev/implementors/features-distribution/#directly-reference-tarball

Comment (joshspicer) [reply proposals/devcontainer-lockfile.md:35]: The spec might need to be updated to support:

`devcontainer-feature-<featureId>.tgz` or `devcontainer-feature-<featureId>-<version>.tgz`

Comment (chrmarti) [reply proposals/devcontainer-lockfile.md:35]: Updating the example. Thanks.

Comment (chrmarti) [reply proposals/devcontainer-lockfile.md:35]: On adding the version to the filename: Not sure that is needed, we would set the "version" attribute to the version in the feature manifest.

**Review by joshspicer — APPROVED** [review]

**Pull request merged**